pkg/controller/nodeipam/ipam/cidrset/metrics.go

```diff
@@ -44,6 +44,15 @@ var (
 		},
 		[]string{"clusterCIDR"},
 	)
+	cidrSetMax = metrics.NewCounterVec(
```

Comment:
In theory, a limit can decrease or increase when it is configurable so it would be better to use a Gauge instead of a Counter.